### DISCUSSION TITLE
Is it possible to hide the App Name in the mac Menu Bar?

### DISCUSSION THREAD

**Discussion by MikeCraig418** [wailsapp/wails#2144] [Q&A] [answered]
For example, when I initiate Alfred, the app appears in the foreground but it doesn't change the name in the Menu Bar. 

In the photo below, "MyApp" is active, but I would like the Menu Bar to say "Chrome" instead.

<img width="456" alt="image" src="https://user-images.githubusercontent.com/23151962/204458098-fd142bf1-7df7-4104-8288-57f956fae97b.png">

Is this possible?

**Comment by stffabi** [comment]
I'm not sure this is possible at all from apple side. I've been using mac quite a while now and have never seen that an app would not show it's name there.
AFAIK this is handled directly by the OS, also changing that to the active app is done from the OS.

But maybe there's some way, have you found some swift/objective-c code to do that? So it might get ported to Wails.

**Reply by OlegGulevskyy** [reply]
Hm, but they are right in one thing though, when Alfred is triggered with hotkey, they don't overwrite the title. It's as if Alfred window is above everything else, like a ghost :)
No idea how this was achieved though

**Reply by stffabi** [reply]
Maybe they are running a service in the background which binds to the hotkeys and that communicates through something like a unix socket with the GUI. So in that case the "Alfred" window is not activated and the name doesn't show up.

**Reply by leaanthony** [reply]
You can use `LSUIElement` in `Info.plist` to achieve this: https://developer.apple.com/documentation/bundleresources/information_property_list/lsuielement?language=objc

**Reply by MikeCraig418** [reply]
> You can use `LSUIElement` in `Info.plist` to achieve this: https://developer.apple.com/documentation/bundleresources/information_property_list/lsuielement?language=objc

@leaanthony - Thanks for sharing this.

I've tried:

```
<key>LSUIElement</key>
<string>true</string>
```
```
<key>LSUIElement</key>
<string>1</string>
```
```
<key>LSUIElement</key>
<true/>
```

in both `Info.plist` and `Info.dev.plist`.

I've confirmed that the resulting binary contains the added line when `Showing Package Contents` and viewing the Info.plist.

Unfortunately, it doesn't appear that Wails supports this? After trying different iterations, I ran `rm -rf build/bin && wails dev`. I also tried `rm -rf build/bin && wails build`. I also tried changing my app name in `wails.json`.

After each attempt, I can still see the icon in my Dock and I can still see the icon when I CMD + Tab.

---

As a side note, I can confirm that 
```
<key>LSUIElement</key>
<string>1</string>
```

works as expected by opening a **different Mac app** in my ~/Applications folder. In this case, I opened _Telegram.app_ (via Show Contents) and then I added the key-pair to the Info.plist. Here, it worked as expected.  My Telegram app is no longer present in my Dock or in CMD + Tab.

It appears that Wails is not allowing this to happen. In my research, I found a [python app](https://github.com/pyinstaller/pyinstaller/pull/1206#issuecomment-154178317) that used `TransformToForegroundApplication` which overrode the functionality of `LSUIElement`. Do you know if there is something on `boot` by Wails that is not allowing this flag to be realized?

I'm happy to make a video of my attempt if that will help.

**Reply by leaanthony** [reply]
It might be something we need to support via Cocoa. It'll be part of the tray menu work that's up next for mac

**Reply by MikeCraig418** [reply]
> It might be something we need to support via Cocoa. It'll be part of the tray menu work that's up next for mac

Thanks @leaanthony, I've only been around here for a few days. I can appreciate how responsive you are.

Do you have a soft-ETA?

**Reply by leaanthony** [reply]
I would expect a PR perhaps in the next month

**Reply by MikeCraig418** [reply]
Thanks @leaanthony

**Comment by MikeCraig418** [comment, marked as answer]
Thank you everyone for your feedback. Thank you, @leaanthony, for assisting and pointing me in the Cocoa direction.

Here's what I ended up doing for the short term:

1. In my project directory, I ran `go mod vendor`
2. I edited `~vendor/github.com/wailsapp/wails/v2/internal/frontend/desktop/darwin/AppDelegate.m`
3. I found `[NSApp setActivationPolicy:NSApplicationActivationPolicyRegular]` and replaced it with `[NSApp setActivationPolicy:NSApplicationActivationPolicyAccessory]`

Running `wails dev` now launches as intended now.

**Reply by leaanthony** [reply]
Glad you have a workaround 👍 You could open a PR to allow a Wails dev to set it if you like....
